lib/data/managers/firestore_manager.dart

```diff
@@ -113,13 +97,59 @@ class FireStoreManager {
     }
     return articlesEntitiesWithImagePath;
   }
+
   Future<String> getImageDownloadURL(DocumentSnapshot imageDocument) async {
-    final sizePath = imageDocument.data["sizes"].first["path"];
-    final imageFileNamePath = imageDocument.data["file"];
+    final sizePath = imageDocument.data[imageSizes].first[imagePath];
+    final imageFileNamePath = imageDocument.data[imageFile];
     final flamelinkPath =
         IMAGE_BASE_PATH + '/' + sizePath + '/' + imageFileNamePath;
     final storageReference =
         FirebaseStorage.instance.ref().child(flamelinkPath);
     return await storageReference.getDownloadURL();
   }
+
+  Future<ArticlesDirectoryEntity> getArticlesDirectory() async {
+    ArticlesDirectoryEntity articlesDirectory;
+
+    // Filters defined by product definition. - ARTICLES DIRECTORY
+    var query = Firestore.instance
+        .collection(FLAME_LINK_CONTENT)
+        .where(FLAME_LINK_SCHEMA, isEqualTo: Schema.ARTICLE_DIRECTORY)
+        .where(FLAME_LINK_SCHEMA_TYPE, isEqualTo: SchemaType.SINGLE)
+        .where(FLAME_LINK_ENVIROMENT,
+            isEqualTo: FirestoreEnvironment.PRODUCTION)
+        .where(FLAME_LINK_LOCALE, isEqualTo: Locale.EN_US);
+
+    await query.getDocuments().then((snapshot) {
+      articlesDirectory = snapshot.documents.map((articlesDirectoryDocument) {
+        return ArticlesDirectoryEntity.featuredArticlesFromDocument(
+            articlesDirectoryDocument);
+      }).first;
+    });
+    return articlesDirectory;
+  }
+
+  Future<ArticlesDirectoryEntity> getFeaturedArticles(
```

Comment:
This method is a bit misleading because is the input and output is the same. In fact the input object is being modified and returned. 

I would suggest to change this so that only the data required by the method is passed and the real data built up is returned. that is the input is a list of article path reference and then the return is a list of articles.

Reply:
You're right it's redundant. Solved.